ote/src/cljs/ote/views/service_viewer.cljs

```diff
@@ -493,32 +531,362 @@
-        pre-booking-data (get-in ts [::t-service/passenger-transportation ::t-service/advance-reservation])
-        booking-data (get-in ts [::t-service/passenger-transportation ::t-service/booking-service])
+  (let [service-sub-type (get ts ::t-service/sub-type)
+        sub-type-key (svc/create-sub-type-key service-sub-type)
+        interfaces (::t-service/external-interfaces ts)
+        luggage-restrictions (get-in ts [sub-type-key ::t-service/luggage-restrictions])
+        real-time-info-data (get-in ts [sub-type-key ::t-service/real-time-information])
+        pre-booking-data (get-in ts [sub-type-key ::t-service/advance-reservation])
+        booking-data (get-in ts [sub-type-key ::t-service/booking-service])
         accessibility-data (:accessibility ts)
         pricing-data (:pricing-info ts)
-        service-hours-data (:service-hours-info ts)]
+        rental-payment-methods (get-in ts [::t-service/rentals ::t-service/payment-methods])
+        rentals (sub-type-key ts)
+        service-hours-info (:service-hours-info ts)
+        vehicle-classes (get-in ts [::t-service/rentals ::t-service/vehicle-classes])
+        vehicle-price-url (get-in ts [::t-service/rentals ::t-service/vehicle-price-url])
+        rental-additional-services (get-in ts [::t-service/rentals ::t-service/rental-additional-services])
+        rental-usage-area (get-in ts [::t-service/rentals ::t-service/usage-area])
+        rental-pick-up-locations (get-in ts [::t-service/rentals ::t-service/pick-up-locations])
+        pick-up-locations-url (get-in ts [::t-service/rentals ::t-service/pick-up-locations-url])]
     (if (or (= (:error to) 404)
             (= (:error ts) 404))
       [:h2 (tr [:common-texts :data-not-found])]
-      [:div
-       [service-header (::t-operator/name to) (::t-operator/id to) (::t-service/id ts)]
-       [operator-info (tr [:service-viewer :operator-info]) to]
-       [service-info (tr [:service-viewer :transport-service-info]) ts]
-       [service-area e! (tr [:service-viewer :service-area]) ts]
-       [published-interfaces (tr [:service-viewer :published-interfaces]) interfaces]
-       [luggage-warnings (tr [:service-viewer :luggage-warnings]) warnings]
-       [real-time-info (tr [:service-viewer :real-time-info]) real-time-info-data]
-       [pre-booking (tr [:service-viewer :advance-reservation]) pre-booking-data]
-       [booking-service (tr [:service-viewer :reservation-service]) booking-data]
-       [accessibility-and-other-services (tr [:service-viewer :accessibility-and-other-services]) accessibility-data]
-       [price-information (tr [:service-viewer :price-information]) pricing-data]
-       [service-hours (tr [:service-viewer :service-hours]) service-hours-data]])))
+      (case service-sub-type
+        :rentals
+        [:div
+         [service-header (::t-operator/name to) (::t-operator/id to) (::t-service/id ts)]
```

Comment:
Having identical definition for a sub-component under each case, one alternative is to declare a local contructor function in the let and call that under each case. That way any updates to the sub-component creation can be done in one place.

Reply:
This is a good idea.
I moved identical components before the case and leave only relevant components.